packages/twenty-server/src/engine/core-modules/code-engine/drivers/lambda.driver.ts

```diff
@@ -0,0 +1,96 @@
+import fs from 'fs';
+
+import {
+  CreateFunctionCommand,
+  Lambda,
+  LambdaClientConfig,
+  InvokeCommand,
+} from '@aws-sdk/client-lambda';
+import { CreateFunctionCommandInput } from '@aws-sdk/client-lambda/dist-types/commands/CreateFunctionCommand';
+
+import { CodeEngineDriver } from 'src/engine/core-modules/code-engine/drivers/interfaces/code-engine-driver.interface';
+
+import { createZipFile } from 'src/engine/core-modules/code-engine/utils/create-zip-file';
+import { BuildDirectoryManager } from 'src/engine/core-modules/code-engine/utils/build-directory-manager';
+import { FunctionMetadataEntity } from 'src/engine/metadata-modules/function-metadata/function-metadata.entity';
+import { FileStorageService } from 'src/engine/integrations/file-storage/file-storage.service';
+import { CommonDriver } from 'src/engine/core-modules/code-engine/drivers/common.driver';
+
+export interface LambdaDriverOptions extends LambdaClientConfig {
+  fileStorageService: FileStorageService;
+  region: string;
+  role: string;
+}
+
+export class LambdaDriver extends CommonDriver implements CodeEngineDriver {
+  private readonly lambdaClient: Lambda;
+  private readonly lambdaRole: string;
+  private readonly fileStorageService: FileStorageService;
+
+  constructor(options: LambdaDriverOptions) {
+    super();
+    const { region, role, ...lambdaOptions } = options;
+
+    this.lambdaClient = new Lambda({ ...lambdaOptions, region });
+    this.lambdaRole = role;
+    this.fileStorageService = options.fileStorageService;
+  }
+
+  async build(functionMetadata: FunctionMetadataEntity) {
+    const javascriptCode = await this.getCompiledCode(
+      functionMetadata,
+      this.fileStorageService,
+    );
+
+    const buildDirectoryManager = new BuildDirectoryManager();
+
+    const {
+      sourceTemporaryDir,
+      lambdaZipPath,
+      javascriptFilePath,
+      lambdaHandler,
+    } = await buildDirectoryManager.init();
+
+    fs.writeFileSync(javascriptFilePath, javascriptCode);
+
+    await createZipFile(sourceTemporaryDir, lambdaZipPath);
+
+    const params: CreateFunctionCommandInput = {
+      Code: {
+        ZipFile: fs.readFileSync(lambdaZipPath),
```

Comment:
Same here. What is the reason behind using a sync I/O operation?

Instinctively, I would use the `node:fs/promises` module to get or write the content of the file.

```suggestion
        ZipFile: await fs.promises.readFile(lambdaZipPath),
```